drivers/ethernet/phy/phy_adin2111.c

```diff
@@ -214,6 +214,36 @@ static int phy_adin2111_reg_write(const struct device *dev, uint16_t reg_addr,
 	return ret;
 }
 
+static int phy_adin2111_reg_read_mmd(const struct device *dev, uint8_t dev_addr, uint16_t reg_addr,
+				     uint32_t *data)
+{
+	const struct phy_adin2111_config *cfg = dev->config;
+	int ret;
+
+	mdio_bus_enable(cfg->mdio);
```

Comment:
Be careful with `mdio_bus_enable|disable` for ADIN. They are used to lock|unlock the MAC.

Reply:
This is a copy of the c22 phy_read/write Implementation for c44. If this is wrong, then the c22 Implementation is also wrong.

Reply:
The problem why a new function is needed due to the conversion of uint16 and uint32 in the phy api.

Reply:
Won't it be better to use `uint16_t data` for the mmd?

Reply:
@manuargue commented to use 32bit to be consistent with the old phy read write api.

Reply:
Maybe it is better not to have mmd api but correct existing c22 to use `uint16_t`? Clause 22 registers are 16 bits wide.

Reply:
The mmd api is needed as you need three addresses to access a register (phy, device, register). 

The general question is if 32bit was the right choice for the old read/writw api to begin  with. Mmds are also only 16bit. My intention to add this was, that I have a device, where the phy ist part of the soc and the api is very handy, to hide the why to access the phy.

I'm fine with removing this change entirety. I was ask to implement a reference, I have a different driver, which a want to publish.

Reply:
> Won't it be better to use uint16_t data for the mmd in the API?

My suggestion to use `uint32_t` was to be consistent with the existent phy reg API, although I agree that it should be better to switch both APIs to `uint16_t`. Doing so will need to have all drivers consuming the API updated accordingly and it should be documented in the migration notes as it's a breaking change.

> The general question is if 32bit was the right choice for the old read/writw api to begin with. Mmds are also only 16bit.

The only reason I can think of it was added as `uint32_t`, is if it was supposed to hold also the MMD in the upper 16-bits, but this is just speculation.

Reply:
> The only reason I can think of it was added as uint32_t, is if it was supposed to hold also the MMD in the upper 16-bits, but this is just speculation.

Now the API allows also a different mapping between mdio and phy registers, where 32bits can be used.

All the implantations if seen in zephyr are wrappers around the mdio functions.

Reply:
> Now the API allows also a different mapping between mdio and phy registers, where 32bits can be used.

Right, I was thinking only in the context of 802.3 protocol, but in fact the `phy_read|write` APIs could be implemented for other management interfaces using 32-bit data (like an in-band host interface for example).

For the MMD API, since it seems to be tied to 802.3 C45, then it makes sense to keep it 16-bit as it was originally proposed.